### DISCUSSION TITLE
How to have a dynamic Subpath along with a base path

### DISCUSSION THREAD

**Discussion by vsvanshi** [vercel/next.js#37577] [Help] [answered]
Hello team,

In my next js application, I have a requirement where along with a base path, i need to have a dynamic subpath after which all my pages should render. Kind of similar to i18n but the query param needs to be controlled from my code.
Any ideas how to achieve this?

I have already tried having `[tenantId]` folder in the pages folder under which i put all of my respective routes. This works correctly until I have to do a client side navigation which completely breaks the path. I have listed this with complete explanation in the issue #37548 

Please any directions would help to solve this problem

**Comment by icyJoseph** [comment, marked as answer]
Please try using the `href + as` combination:

```jsx
<Link href="/[tenantId]/path" as={`/${router.query.tenantId}/jobdetails/12312`} />
```

Where `href` props reflects the file system routing, and the `as` prop reflects what is shown in the browser.

Let me know how it goes.

**Reply by vsvanshi** [reply]
Thanks for the quick reply @icyJoseph .

Figured out the exact problem. I was missing a starting "/" in the Link href in both conditions that was making it wrong, which is also specified in your answer.
However, using either "href" or "href + as" behave similar for this problem if we are not using the starting slash.

**Reply by icyJoseph** [reply]
Ah right! the tenantId must be preceded by the slash, updating answer.
